contracts/libs/Ibc.sol

```diff
@@ -305,13 +305,16 @@ library Ibc {
         returns (address emittingContract, bytes[] memory topics, bytes memory unindexedData)
     {
         // The first byte is a RLP encoded receipt type so slice it off.
-        RLPReader.RLPItem[] memory receipt;
-        if (receiptRLP.length % 2 == 0) {
-            // If even, we don't clip bytes
-            receipt = RLPReader.readList(receiptRLP);
-        } else {
-            receipt = RLPReader.readList(Bytes.slice(receiptRLP, 1, receiptRLP.length - 1));
+        uint8 typeByte;
+        assembly {
+            typeByte := byte(0, mload(add(receiptRLP, 32)))
```

Comment:
```suggestion
            typeByte := byte(0, mload(add(receiptRLP, 1)))
```

if byte is u8 can we get away with only loading 1 byte?

Reply:
or maybe i'd try `typeByte := mload(add(receiptRLP, 1))`?

Reply:
i believe its the first byte after the result of mload which needs to skip the first 32 bytes (length) of the receiptRLP which is an RLPItem (i think) in the form {length, ptr}